llvm/utils/TableGen/InstrInfoEmitter.cpp

```diff
@@ -1316,6 +1394,14 @@ void InstrInfoEmitter::emitEnums(
   }
   OS << "    INSTRUCTION_LIST_END = " << NumberedInstructions.size() << '\n';
   OS << "  };\n";
+
+  if (!Target.getAllRegClassByHwMode().empty()) {
```

Comment:
Should we put `Target.getAllRegClassByHwMode()` in a variable instead of calling it twice?